File: SupplierPortal.Web/Modules/ServerTypes/Items/ItemForm.ts
import { IntegerEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ItemForm {
    ItemId: IntegerEditor;
    ItemName: StringEditor;
}

export class ItemForm extends PrefixedContext {
    static readonly formKey = 'Items.Item';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ItemForm.init)  {
            ItemForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(ItemForm, [
                'ItemId', w0,
                'ItemName', w1
            ]);
        }
    }
}

File: SupplierPortal.Web/Modules/Items/Item/ItemForm.cs
using Serenity.ComponentModel;

namespace SupplierPortal.Items.Forms;

[FormScript("Items.Item")]
[BasedOnRow(typeof(ItemRow), CheckNames = true)]
public class ItemForm
{
    public int ItemId { get; set; }
    public string ItemName { get; set; }
}